### ISSUE TITLE
Copying folder sometimes doesn't finish completely

### ISSUE THREAD

**Issue by tobihagemann** [cryptomator/ios#173]
### Please agree to the following

- [X] I have searched [existing issues](https://github.com/cryptomator/ios/issues?q=) for duplicates
- [X] I agree to follow this project's [Code of Conduct](https://github.com/cryptomator/ios/blob/develop/.github/CODE_OF_CONDUCT.md)

### Summary

Copying a folder into the vault sometimes doesn't finish completely. It doesn't "fail" per se but some subfolders may be inaccessible or files aren't uploaded.

### System Setup

- iOS: 15.3.1
- Cryptomator: 2.1.1 (830)

### Cloud Type

Other File Provider

### Steps to Reproduce

1. Create a new vault (e.g., into "On My iPhone")
2. Copy the encrypted vault (or any other folder but this was the easiest to reproduce if you don't have a folder ready with a certain hierarchy)
3. Unlock vault
4. Paste folder from step 2

### Expected Behavior

Folder should be copied completely.

### Actual Behavior

Since it can't always be reproduced, the actual behavior varies.

- It may have worked. Therefore, all subfolders and files are accessible.
- Some subfolders may be broken and show a "Content Unavailable" error.
- Some files may be stuck in "Waiting…".

### Reproducibility

Intermittent

### Relevant Log Output

```shell
importDocument uploadFile failed: Error Domain=NSFileProviderErrorDomain Code=-1005 "The file doesn’t exist."
```

**Comment by sindastra** [comment]
Hi there! 👋

I came here because I'm having the same issue. I copied two folders, each containing two files. Both folders were created and synced, but only the files in one of the folders got uploaded/synced. The files in the other folder are stuck at "Waiting...".

If you know how to force/reset/flush this, please let me know.

### A bit of general feedback:

I was actually excited when I saw there was a new Cryptomator app (yes, really :D), but when I saw it was using the "Files" app exclusively, I was a bit skeptical. As in, whether this will work with my workflow/features missing and if this would be stable. However, I thought about it, and the Files app integration makes sense and has its advantages. So, I decided to migrate to it yesterday, but I'm encountering many issues (like this one here), and I have to say, it seems it's overall "not quite ready yet".

I guess what I'm trying to say is: Please keep the original Cryptomator app in the App Store, at least for the foreseeable future/until things are ironed out. As it stands, I'm a bit concerned, and wouldn't want to miss Cryptomator on the go (I noticed the original app's repo is marked as archived/EOL).

But I'd also like to say the concept makes sense, and I'm actually looking forward to migrating to the new app once it's stable, but for now, sadly I'll have to use the original one. (:

###### I hope this sort of general feedback is OK, I thought it "fits here".

**Comment by tobihagemann** [comment]
As far as I know, the only way to really "reset" things is to remove the vault from the vault list and add it again. Be aware that files that could not be uploaded may get lost.

General feedback doesn't quite fit here since it's off-topic but I understand your concerns and we'll keep Cryptomator 1.x in the App Store for now until all major issues have been addressed. To be fair, Cryptomator 2 was in a long beta phase and based on the feedback, it was ready for a final release. As with all software, issues rise up eventually and will be fixed with updates.

**Comment by sindastra** [comment]
Hi there, I just tried the latest update (2.1.3, 844), for which the PR reads that it "might" solve this issue here.

I just attempted to upload a single file, and it's stuck at "Waiting...", in the same way as with the folder copying issue. So, I thought I'd report it here, as it seems related.

So, after quick testing, it seems the update did unfortunately not solve this particular issue here.

**Comment by tobihagemann** [comment]
Yes, the issue fixed in 2.1.3 was a different one. Basically, the upload didn't even get started and aborted directly with an error.

We'll address this issue next. We've already made some progress and found the cause. A fix may take some time though, it's a nasty race condition.
